### ISSUE TITLE
Where/how do users download the data?

### ISSUE THREAD

**Issue by albenson-usgs** [gbif/hosted-portals#23]
I know in the call for proposals it was stated that a redirect from the search results would send people to GBIF.org for data downloads but I can't find where that is in the portal.

**Comment by MortenHofft** [comment]
Hy Abby - it isn't there yet. But yes, that is the plan.

**Comment by tucotuco** [comment]
Hi @MortenHofft. This functionality is all important for VertNet. We can't really even have people try using the hosted portal without a download solution in place. Is there somewhere we can track progress on this?

**Comment by MortenHofft** [comment]
On this issue :)

**Comment by MortenHofft** [comment]
In case anyone is interested in why this isn't as simple as adding a button:

One of the challenges in downloads is that the API used to query on the hosted portals are much richer than the current search API. In some aspects richer than the download API as well. And in some cases just slightly different. E.g. https://github.com/gbif/occurrence/issues/245

Another example is that the hosted portals allow for negated queries on e.g. issues. This is not allowed in API v1. But for all of this improvements we have to make sure that they behave the same when downloading, and that is not the case currently.

**Comment by MortenHofft** [comment]
For those who wish to follow on a commit basis, then these are the more relevant repositories
* https://github.com/gbif/jekyll-hp-base-theme - everything related to the Jekyll theme (everything but data exploration)
* https://github.com/gbif/gbif-web - data exploration UI for the hosted portals
* https://github.com/gbif/pipelines and https://github.com/gbif/occurrence - things related to occurrence search and download in a wider sense than just the hosted portals.

**Comment by MortenHofft** [comment]
This is the intended download flow

![download flow](https://user-images.githubusercontent.com/6759267/114540300-a20f5880-9c55-11eb-90ad-77d8aeb81151.jpg)

My main concern is how to keep the user motivated and comfortable throughout this. I think this could work, my only improvement ideas are really

* Have it more in the style of the hosted portal you come from
* Have a custom greeting like "You are about to download data from [hosted portal name]"

What do others think? Would this be a good user experience? Is the wording somewhat what we aim for? I've made room for an about tab/text on the GBIF.org site, but I'm not sure what it should say nor if it is needed.

**Comment by albenson-usgs** [comment]
I am worried about the number of clicks users will need to make to get to the data. I'm guessing having a GBIF login in the portals themselves would be difficult to add and possibly confusing for users? Seeing that first screen makes me wonder about the utility of the free text search. I do understand that it's not possible to download the data based on a free text filter but I'm wondering who would want to use it given that's the case and if it should be displayed first. Basically I'm wondering if we will have a lot of frustrated users thinking "Why is this here if I can't download the data?". I think it might be worth bumping up https://github.com/gbif/hosted-portals/issues/20 and https://github.com/gbif/hosted-portals/issues/107 because my preference would be to move the free text search into the More tab to prevent frustration with not being able to download the data that way. 

The wording you have looks good to me. I'm not sure if we need the About tab on the download page.

**Comment by MortenHofft** [comment]
**Not being able to download free text queries**
This is the same as GBIF.org

Search and download are generated from different (but synced) databases. They have different merits. For structured filters we can translate between the 2, but for free text queries all sorts of word stemming, fuzzy matching etc goes on that cannot be mirrored in the database that is used for downloads. It is annoying but unlikely to change.

**Comment by albenson-usgs** [comment]
Disclaimer that I'm not a usability expert, just a biologist, so please take any of my suggestions with that grain of salt. 

**Not being able to download free text queries**
Completely understand not being able to download free text queries and I don't have any expectation that this would change. Apologies if that was not clear in my previous comment. My issue is more with how to communicate this with a user before they try to download data and that the placement of the keyword search next to filters that are all downloadable filters (right?) might be confusing.
<img width="548" alt="portal_filters" src="https://user-images.githubusercontent.com/7586150/114571426-e5f16600-9c33-11eb-9a7e-8f2fffe751c1.PNG">
If as a user I go to a portal and spend time making various selections and get the data just the way I want it and then I'm told that one of my filters is unsupported for download I would be frustrated that I have to figure out how to reconfigure the download, possibly? 

**Number of clicks**
Makes sense. I don't have strong opinions either way on whether we need the intermediary. I'm ok with keeping it. I will differ to others on this topic.

**Comment by tucotuco** [comment]
I want to support all the observations that Abby has made. In general,
simple is better, confusing is harmful.

**Comment by MortenHofft** [comment]
Thanks @albenson-usgs 

I believe that the download flow can be discussed independently of the free text filter (I get that I triggered your observation with my image). You are welcome to create a separate issue for the problem of free text search not being downloadable. Or for the interface being confusing - ideally with a suggestion for how to solve it. 

For what it is worth, I'm not super exited about that filter either. It is of course a shame that I cannot download, but more importantly it gives misleading results as users mistake it for a scientific filter (they typically enter a taxon name - or at least they did last time I did looked). So I wouldn't configure it as a primary/default filter (unlike the current iteration of GBIF.org where it is the top most).

**Comment by MortenHofft** [comment]
It would be nice to progress this issue, but I'm not sure how to. 

**Redirect or not**
It has been the plan from the start to redirect. From the presentation material introducing the hosted portals it says

> A redirect from search results, sending users to GBIF.org for data downloads, ensuring that the downloads receive DOIs linking back to the datasets included

That isn't to say that we cannot explore alternatives, but it isn't a new idea.

**Free texts queries not downloadable**
This is beyond the work on the hosted portals. If it is considered critical the only options are to
* [remove the free text filter](https://hp-theme.gbif-staging.org/data-exploration-config#excluded-filters)
* [log a feature request](https://github.com/gbif/occurrence/issues/new) to have free text available in downloads 

**Number of clicks**
There are 4 steps in above flow, I'm not sure how to make it simpler:

* Step 1: The user has to remove free text filters if any are applied.
* Step 2: Once the filter is removed it would be odd to redirect the user before telling them that this is about to happen.
* Step 3: The user has to be logged in, as that is a requirement for the GBIF API (see also [meeting May 2021 at 48:31](https://vimeo.com/545392585/7981064d06#t=48m30s))
* Step 4: Select a download format - we can hardly skip this.

It could be made slightly simpler by allowing logins from the hosted portals themselves. That would remove step 2.

And of course it would be simpler if there was no login required and free text download was supported, but that is not something that will change soon.

**Login from the hosted portals**
To do so would be new development. I would have to think through and discuss the security details, but I assume that the API could be opened to allow other sites to login. 

From the UI it would require new development of: login, registration, forgot password, user profiles, welcome pages, email confirmation pages, download summary tree, download type select, download pages? etc. It would also require a new backend to support this. It isn't entirely clear where to make the cut, but my gut feeling is that this is a larger piece of work.

**Here and now**
If I was to decide, then I would stick with the proposed solution. That doesn't mean that it cannot be revisited, but if we want downloads soon, then there is a simple solution ready. But obviously I do not want to force it on anyone either. I guess I had hoped for minor corrections within the proposed solution 😄, but I suppose that is the joy and danger of asking for feedback.

**Comment by tucotuco** [comment]
I support the pursuit of downloads sooner than later, with the restrictions
and steps enumerated. Having the downloads is absolutely necessary for many
portals before they can be released (and further evaluated). Any
simplifications in the future will also be improvements and seen as such by
users, so seems a positive path to take.

**Comment by albenson-usgs** [comment]
First to state - yes agree to just moving forward with "as is" because we need download in place sooner than later. I didn't intend for my comments to be a blocker. Just trying to think things through and figure out if any easy adjustments are possible. Seems like there are not easy adjustments. That's ok! Ability to download takes precedence. Also the keyword search issue was handled somewhat by the work on https://github.com/gbif/hosted-portals/issues/107 so to me that has been "resolved".

**Comment by tucotuco** [comment]
We're also happy to proceed with the keyword search solution as is. Thanks for the reminder @albenson-usgs.

**Comment by MortenHofft** [comment]
Thank you both

I've started this work and added it to the sites, but there are still work left before it is ready to use. I will close this issue once above solution is in place and works. I have created https://github.com/gbif/hosted-portals/issues/161 to capture the discussed future improvements. Thank you for your help.
